File: hinos_web/src/types/hymn.ts
export interface Hymn {
  id: string;
  number: string;
  title: string;
  category: 'hinario' | 'canticos' | 'suplementar' | 'novos';
  hymnBook: string;
  verses: Verse[];
  key?: string;
  bpm?: number;
}

export interface Verse {
  type: 'V1' | 'V2' | 'V3' | 'V4' | 'R' | 'Ponte' | 'C';
  lines: string[];
}

export type CategoryFilter = 'todos' | 'hinario' | 'canticos' | 'suplementar' | 'novos';

File: hinos_api/Models/Hymn.cs
namespace hinos_api.Models;

public class Hymn
{
    public int Id { get; set; }
    public string Number { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty; // hinario, canticos, suplementar, novos
    public string HymnBook { get; set; } = string.Empty;
    public string? Key { get; set; }
    public int? Bpm { get; set; }
    public List<Verse> Verses { get; set; } = new();
}
